mountpoint-s3/src/fs.rs

```diff
@@ -992,12 +992,20 @@ where
                 .ok_or_else(|| err!(libc::EBADF, "invalid directory handle"))?
         };
 
+        // special case where we need to rewind and restart the streaming
+        if offset == 0 && dir_handle.offset() != 0 {
+            // only do this if this is not the first call with offset 0
+            dir_handle.offset.store(0, Ordering::SeqCst);
```

Comment:
nit: put this in a method on `DirHandle` to match its `next_offset()` method

Reply:
Done!